### PR TITLE
updrage to keras 3.0 for metric learning example

### PR DESCRIPTION
**Pull request by madhusshivakumar**
Upgrade Keras to the Latest Version and Adapt the Code for Compatibility

In this commit, I have upgraded the Keras library to its latest version to take advantage of new features and improvements. Additionally, I have made necessary code modifications to maintain the previous functionality while utilizing the updated Keras API.

Key Changes:

Upgraded Keras to version 3.0
Updated deprecated Keras functions to their latest equivalents.
Tested the code to verify that it functions correctly with the new Keras version.
These changes aim to keep our project up to date with the latest tools and technologies while preserving the integrity of our existing functionality.

colab link for reference:

https://colab.research.google.com/drive/1iCM-yB_JpgRS0EdvD6CUjDrNhGN4Lwbs?usp=sharing

### CODE REVIEW

**Comment by madhusshivakumar** [comment]
Currently facing this issue #18698 . will move this PR to draft until the issue is fixed.

File: examples/keras_io/vision/metric_learning.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+`(anchor, positive)` pairs spread across the classes. The goal of learning will be to
+move the anchor and positive pairs closer together and further away from other instances
+in the batch. In this case the batch size will be dictated by the number of classes; for
+CIFAR-10 this is 10.
+"""
+
+num_classes = 10
+
+
+
```
Comment (divyashreepathihalli) [inline examples/keras_io/vision/metric_learning.py:122]: remove extra line

File: examples/keras_io/vision/metric_learning.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+"""
+Finally we run the training. On a Google Colab GPU instance this takes about a minute.
+"""
+
+model.compile(
+    optimizer=keras.optimizers.Adam(learning_rate=1e-3),
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+)
+
+# history = model.fit(AnchorPositivePairs(num_batches=1000), epochs=20)
```
Comment (divyashreepathihalli) [inline examples/keras_io/vision/metric_learning.py:228]: remove the comment and fix the number of epochs = 20

File: examples/keras_io/vision/metric_learning.py [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    "Horse",
+    "Ship",
+    "Truck",
+]
+disp = ConfusionMatrixDisplay(confusion_matrix=confusion_matrix, display_labels=labels)
+disp.plot(include_values=True, cmap="viridis", ax=None, xticks_rotation="vertical")
+plt.show()
+
+"""
+Example available on HuggingFace.
```
Comment (divyashreepathihalli) [inline examples/keras_io/vision/metric_learning.py:318]: remove this section about hugging face and the table

File: examples/keras_io/vision/metric_learning.py [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+For a more detailed overview of metric learning see:
+
+* [What is metric learning?](http://contrib.scikit-learn.org/metric-learn/introduction.html)
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+"""
+
```
Comment (divyashreepathihalli) [inline examples/keras_io/vision/metric_learning.py:27]: if this example only works with TF backend you might want to add a section about setting the `KERAS_BACKEND` variable
```
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
```

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]

**Comment by divyashreepathihalli** [comment]
@madhusshivakumar thanks for the PR! I have requested a few changes to the PR.

File: examples/keras_io/vision/metric_learning.py (hunk 4 above)
Comment (madhusshivakumar) [reply examples/keras_io/vision/metric_learning.py:27]: It works with jax as well, I'll add a section on setting the backend👍🏻

Comment (divyashreepathihalli) [reply examples/keras_io/vision/metric_learning.py:27]: unless we can completely remove `import tensorflow` from our code this would just be TF backend compatible only.

Comment (madhusshivakumar) [reply examples/keras_io/vision/metric_learning.py:27]: Done.

File: examples/keras_io/vision/metric_learning.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+from keras.datasets import cifar10
+
+"""
+## Backend for the new keras
+
+Set Keras backend to tensorflow. 
+"""
+os.environ["KERAS_BACKEND"] = "tensorflow"
```
Comment (divyashreepathihalli) [inline examples/keras_io/vision/metric_learning.py:53]: the backend should be set before `import keras`, so I suggest adding it under the setup section

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]

File: examples/keras_io/vision/metric_learning.py (hunk 1 above)
Comment (madhusshivakumar) [reply examples/keras_io/vision/metric_learning.py]: done

File: examples/keras_io/vision/metric_learning.py (hunk 3 above)
Comment (madhusshivakumar) [reply examples/keras_io/vision/metric_learning.py]: done

File: examples/keras_io/vision/metric_learning.py (hunk 2 above)
Comment (madhusshivakumar) [reply examples/keras_io/vision/metric_learning.py]: done

File: examples/keras_io/vision/metric_learning.py (hunk 5 above)
Comment (madhusshivakumar) [reply examples/keras_io/vision/metric_learning.py:53]: done.

File: examples/keras_io/vision/metric_learning.py [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+* [What is metric learning?](http://contrib.scikit-learn.org/metric-learn/introduction.html)
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+
+Set Keras backend to tensorflow. 
+"""
+os.environ["KERAS_BACKEND"] = "tensorflow"
```
Comment (divyashreepathihalli) [inline examples/keras_io/vision/metric_learning.py:29]: you would need to `import os` before using it here

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]

**Review by divyashreepathihalli — APPROVED** [review]

File: examples/keras_io/vision/metric_learning.py [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+
+Set Keras backend to tensorflow. 
+"""
+import os
+os.environ["KERAS_BACKEND"] = "tensorflow"
```
Comment (fchollet) [inline examples/keras_io/vision/metric_learning.py:30]: If the example only works with TF, please move it to the `tensorflow` folder.

**Review by fchollet — APPROVED** [review]
